dislib/cluster/daura/base.py

```diff
@@ -0,0 +1,106 @@
+from itertools import chain
+
+from pycompss.api.api import compss_wait_on
+from pycompss.api.parameter import COLLECTION_IN, INOUT
+from pycompss.api.task import task
+from sklearn.base import BaseEstimator
+
+import numpy as np
+
+
+class Daura(BaseEstimator):
+
+    def __init__(self, cutoff):
+        """ Perform Daura clustering.
+
+        See:
+        lockhartlab.squarespace.com/blog/2018/1/14/clustering-with-daura-et-al
+
+        Parameters
+        ----------
+        cutoff : float
+            Distance to determine the neighbors of a sample.
+        """
+        self.cutoff = cutoff
+
+    def fit_predict(self, distances):
+        """ Compute Daura clustering.
+
+        Parameters
+        ----------
+        distances : ds-array (n_samples, n_samples)
+            Pairwise distances between the samples.
+
+        Returns
+        -------
+        clusters : List[ List[ int ] ]
```

Comment:
Why not return a ds-array?

Reply:
I opened an issue (#311) to solve this later.